raiden/network/transport/matrix/rtc/web_rtc.py

```diff
@@ -360,99 +390,79 @@ async def reset(self) -> None:
-            "call_id": rtc_partner.call_id,
+            "call_id": self.call_id,
         }
-        self._send_signaling_message(partner_address, json.dumps(message))
+        self._signaling_send(self.partner_address, json.dumps(message))
 
-    def _handle_sdp_callback(
-        self, rtc_session_description: Optional[RTCSessionDescription], partner_address: Address
+    def send_hangup_message(self) -> None:
+        hangup_message = {
+            "type": _RTCMessageType.HANGUP.value,
+            "call_id": self.call_id,
+        }
+        self._signaling_send(self.partner_address, json.dumps(hangup_message))
+
+    def handle_sdp_callback(
+        self, rtc_session_description: Optional[RTCSessionDescription]
     ) -> None:
         """
-        This is a callback function to process sdp (session description protocol) messages.
+        This is a callback function to process SDP (Session Description Protocol) messages.
         These messages are part of the ROAP (RTC Offer Answer Protocol) which is also called
-        signalling. Messages are exchanged via the partners' private matrix room.
+        signalling. Messages are exchanged via Matrix.
         Args:
             rtc_session_description: sdp message for the partner
-            partner_address: Address of the partner
         """
         if rtc_session_description is None:
             return
 
-        rtc_partner = self.get_rtc_partner(partner_address)
-
         sdp_type = rtc_session_description.type
         message = {
             "type": sdp_type,
             "sdp": rtc_session_description.sdp,
-            "call_id": rtc_partner.call_id,
+            "call_id": self.call_id,
         }
         self.log.debug(
             f"Send {sdp_type} to partner",
-            partner_address=to_checksum_address(partner_address),
+            partner_address=to_checksum_address(self.partner_address),
             sdp_description=message,
         )
 
-        self._send_signaling_message(partner_address, json.dumps(message))
+        self._signaling_send(self.partner_address, json.dumps(message))
 
-    def _maybe_initialize_web_rtc(self, address: Address) -> None:
-        if address in self._web_rtc_channel_inits:
-            return
 
-        lower_address = my_place_or_yours(self.node_address, address)
+@dataclass
+class WebRTCManager(_CoroutineHandler, Runnable):
+    def __init__(
+        self,
+        node_address: Address,
+        process_messages: Callable[[List[ReceivedRaidenMessage]], None],
+        signaling_send: Callable[[Address, str], None],
+        stop_event: GEvent,
+    ) -> None:
+        super().__init__()
+        self.node_address = node_address
+        self._process_messages = process_messages
+        self._signaling_send = signaling_send
+        self._stop_event = stop_event
+        # the addresses for which we're currently trying to initialize WebRTC channels
+        self._web_rtc_channel_inits: Set[Address] = set()
```

Comment:
Nitpick: If we rename the variable for something else, do you think we could lose the comment? I personally think that something like `self._web_rtc_channels` communicates this idea pretty well.

Reply:
I'm not sure that that would be more appropriate -- note these are not channels, but the addresses of channels we're trying to establish. Most of the time this set is expected to be empty, because all the channels are initialized. It is only _during initialization_ that an address shows up in that set. Hence the `_inits` suffix.